Detectors/FIT/common/simulation/src/Digitizer.cxx

```diff
@@ -18,26 +18,88 @@
 #include <algorithm>
 #include <cassert>
 #include <iostream>
+#include <optional>
 
 using namespace o2::fit;
 //using o2::fit::Geometry;
 
 ClassImp(Digitizer);
 
-double get_time(const std::vector<double>& times, double signal_width)
+double sinc(const double x)
 {
-  TH1F hist("time_histogram", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
-  auto signalForm = [](double x) {
-    return -(exp(-0.83344945 * x) - exp(-0.45458 * x));
-  };
-  for (auto time : times)
-    for (int bin = hist.FindBin(time); bin < hist.GetSize(); ++bin)
-      if (hist.GetBinCenter(bin) > time)
-        hist.AddBinContent(bin, signalForm(hist.GetBinCenter(bin) - time));
-  double maximum{ hist.GetMaximum() };
-  int binfound{ hist.FindFirstBinAbove(maximum * 0.4) };
-  //  std::cout<<"@@@@@!!! max "<<maximum<<" binfound "<<binfound<<" return "<<hist.GetBinCenter(binfound)<<std::endl;
-  return hist.GetBinCenter(binfound);
+  if (x == 0)
+    return 1;
+  return sin(x) / x;
+}
+double signalForm_i(double x)
+{
+  return -(exp(-0.83344945 * x) - exp(-0.45458 * x)) * (x >= 0) / 7.8446501; // Maximum should be 7.0/250 mV
+};
+
+double Digitizer::get_time(const std::vector<double>& times, double signal_width)
+{
+  mHist->Reset();
+  mHistsum->Reset();
+  mHistshift->Reset();
+  /*
+  TH1F mHist("time_mHistogram", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
+  TH1F mHistsum("time_sum", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
+  TH1F mHistshift("time_shift", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
+  */
+  /// Fill MHistrogram `mHist` with photoelectron induced voltage
+  for (auto time : times) {
+    for (int bin = mHist->FindBin(time); bin < mHist->GetSize(); ++bin)
+      if (mHist->GetBinCenter(bin) > time)
+        mHist->AddBinContent(bin, signalForm_i(mHist->GetBinCenter(bin) - time));
+  }
+  /// Add noise to `mHist`
+  double noiseVar = 0.1;
+  int noise_period = 1 / 0.9 / mHist->GetBinWidth(0);
+  for (int c = 0; c < mHist->GetSize(); c += noise_period) {
+    double val = gRandom->Gaus(0, noiseVar);
+    for (int bin = 0; bin < mHist->GetSize(); ++bin)
+      mHist->AddBinContent(bin, val * sinc(TMath::Pi() * (bin - c) / (double)noise_period));
+  }
+  int binshift = int(1.47 / (signal_width / 1000.));
+  /// Add noise to the initial part of `mHistshift` that would contain values before the timeframe
+  for (int c = 0; c < mHist->GetSize(); c += noise_period) {
```

Comment:
Perhaps these 2 outer for loops can be merged to one? In fact, this random ``val``, should not it be the same in both loops. Also, to minimize the number of divisions, it is better to precalculate once ``double noisePeriodInv=1./noise_period;`` and multiply by it.